### PR TITLE
LASB-2827 New assessor-details endpoint to support Application Tracking Service

### PR DESCRIPTION
**Pull request by stone-dj**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-2827)

Changes include:
- New endpoint `/api/internal/v1/assessment/rep-orders/{repId}/ioj-assessor-details` defined in `RepOrderController`
- Functionality to retrieve details of the interests of justice assessor for a given representation order

Local testing was done via Postman

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

**Comment by lkm0287** [comment]
Be a bit careful putting screenshots in these PRs with any data - they're publicly accessible.

File: README.md [hunk 1]
```diff
@@ -44,12 +45,22 @@ Once the steps have been completed you can decrypt your local copy of the reposi
...
 
- You will need to have the relevant database accessible on port 1521 locally. This can be provided by an SSH tunnel to an RDS instance in AWS. Here is the command to tunnel to Dev (add your user Bastion user name):
+You will need to have the relevant database accessible on port 1521 locally. This can be provided by an SSH tunnel to an RDS instance in AWS.  
+The steps required to configure access via Bastions are listed [here](https://dsdmoj.atlassian.net/wiki/spaces/aws/pages/4584865935/AWS+-+).  
+The command to create an SSH tunnel (with port forwarding for 1521) to Dev is included below.  
+*Remember to replace _**astone**_ with your own Bastion username.
 
 ```sh
-ssh -L 1521:rds.maat.aws.dev.legalservices.gov.uk:1521 <username>@35.176.251.101 -i ~/.ssh/id_rsa
+ssh -vvv -L 1521:rds.maat.aws.tst.legalservices.gov.uk:1521 astone@ssmbastion-non-prod-lz
```
Comment (lkm0287) [inline README.md:54]: Same comment about these endpoints - should stuff like this be public?

**Comment by stone-dj** [comment]
> Be a bit careful putting screenshots in these PRs with any data - they're publicly accessible.

Fair point, I'll remove it as the test data is based of prod data.

File: README.md (hunk 1 above)
Comment (stone-dj) [reply README.md]: agreed, I have updated to be a bit more limited with the information shared.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/repository/RepOrderRepository.java [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 
 @Repository
 public interface RepOrderRepository extends JpaRepository<RepOrderEntity, Integer>, JpaSpecificationExecutor<RepOrderEntity> {
 
+    @Query(value = """
```
Comment (lkm0287) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/repository/RepOrderRepository.java:14]: Why is there three punctuation marks around the value?

Comment (lkm0287) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/repository/RepOrderRepository.java:14]: I'm guessing it's a multi-line string?

File: maat-court-data-api/src/main/java/gov/uk/courtdata/repository/RepOrderRepository.java [hunk 3]
```diff
@@ -1,12 +1,29 @@
...
+                   NVL(INITCAP(U.FIRST_NAME
+                               || ' '
+                               || U.SURNAME),
+                       R.USER_CREATED) name,
+                       R.USER_CREATED userName
+               FROM
+                   REP_ORDERS R
+                   LEFT JOIN USERS U ON (U.USER_NAME = R.USER_CREATED)
+               WHERE
+                   ID = :repOrderId
```
Comment (lkm0287) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/repository/RepOrderRepository.java:25]: I think at least one index would be beneficial for this query - just checked production and there is no indexing on the left join field.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/repository/RepOrderRepository.java (hunk 2 above)
Comment (stone-dj) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/repository/RepOrderRepository.java:14]: yes, it is a feature that came with Java 15, I felt it made the query significantly easier to read compared to the alternatives.

Like:
```
    @Query(value = "   SELECT" +
                   "       NVL(INITCAP(U.FIRST_NAME" +
                   "                   || ' '" +
                   "                   || U.SURNAME)," +
                   "           R.USER_CREATED) name," +
                   "           R.USER_CREATED userName" +
                   "   FROM" +
                   "       REP_ORDERS R" +
                   "       LEFT JOIN USERS U ON (U.USER_NAME = R.USER_CREATED)" +
                   "   WHERE" +
                   "       ID = :repOrderId",
```
Or:
```
    @Query(value = " SELECT NVL(INITCAP(U.FIRST_NAME || ' ' || U.SURNAME), R.USER_CREATED) name, R.USER_CREATED userName FROM REP_ORDERS R LEFT JOIN USERS U ON (U.USER_NAME = R.USER_CREATED) WHERE ID = :repOrderId",
            nativeQuery = true)
```

https://www.baeldung.com/java-text-blocks

**Review by venkat980 — APPROVED** [review]

**Pull request merged**